v0/destinations/attentive_tag/transform.js

```diff
@@ -0,0 +1,123 @@
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  getIntegrationsObj,
+  CustomError
+} = require("../../util");
+
+const identifyResponseBuilder = (message, { Config }) => {
+  const { apiKey, signUpSourceId } = Config;
+  let endpoint; 
+  const integrationsObj = getIntegrationsObj(message, "attentive_tag");
+  let payload;
+  if (integrationsObj) {
+    if (integrationsObj.signUpSourceId) {
+      signUpSourceId = integrationsObj.signUpSourceId;
+    }
+    if (integrationsObj?.identifyOperation.toLowerCase() === "unsubscribe") {
+      endpoint = "/subscriptions/unsubscribe";
+      payload = constructPayload(message, attentiveUnsubscribeConfig);
+    }
+    if (integrationsObj?.identifyOperation.toLowerCase() == "subscribe") {
+      endpoint = "/subscriptions";
+      payload = constructPayload(message, attentiveSubscribeConfig);
+    }
+  }
+  if (payload) {
+    const response = defaultRequestConfig();
+    response.endpoint = `${BASE_URL}endpoint`;
+    response.headers = {
+      Authorization: `Bearer ${apiKey}`,
+      "Content-Type": "application/json"
+    };
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+  }
+  return response;
+};
+const trackResponseBuilder = (message, { Config }) => {
+  const { apiKey } = Config;
+
+  let event = get(message, "event");
+  let endpoint;
+
+  switch (event.toLowerCase()) {
+    /* Browsing Section */
+    case "product_list_viewed":
+      payload = constructPayload(
+        message,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.PRODUCT_LIST_VIEWED.name]
+      );
+      endpoint = CONFIG_CATEGORIES.PRODUCT_LIST_VIEWED.endpoint;
+      break;
+    /* Ordering Section */
+    case "product_viewed":
+      payload = constructPayload(
+        message,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.PRODUCT_VIEWED.name]
+      );
+      endpoint = CONFIG_CATEGORIES.PRODUCT_VIEWED.endpoint;
+      break;
+    case "order_completed":
+      payload = constructPayload(
+        message,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.ORDER_COMPLETED.name]
+      );
+      endpoint = CONFIG_CATEGORIES.ORDER_COMPLETED.endpoint;
+      break;
+    case "product_added":
+      payload = constructPayload(
+        message,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.PRODUCT_ADDED.name]
+      );
+      endpoint = CONFIG_CATEGORIES.PRODUCT_ADDED.endpoint;
+      break;
+    default:
+      payload = constructPayload(
+        message,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.TRACK.name]
+      );
+      endpoint = CONFIG_CATEGORIES.TRACK.endpoint;
+  }
+  if (payload) {
+    const response = defaultRequestConfig();
+    response.endpoint = `${BASE_URL}${endpoint}`;
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.headers = {
+      Authorization: `Bearer ${apiKey}`,
+      "Content-Type": "application/json"
+    };
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+    return response;
```

Comment:
line 88-97 is repetitive with line no 32-40. Move it to a separate function and write once.

Reply:
Done